llama_hub/tools/ondemand_loader_tool.py

```diff
@@ -0,0 +1,138 @@
+"""Ad-hoc data loader tool.
+
+Tool that wraps any data loader, and is able to load data on-demand.
+
+"""
+
+
+from llama_index.tools.types import BaseTool, ToolMetadata
+from llama_index.readers.base import BaseReader
+from typing import Any, Optional, Dict, Type
+from llama_index.indices.base import BaseIndex
+from llama_index.indices.vector_store import VectorStoreIndex
+from llama_index.tools.utils import create_schema_from_function
+from pydantic import BaseModel
+from llama_index.tools.function_tool import FunctionTool
+
+
+class OnDemandLoaderTool(BaseTool):
```

Comment:
for this and the load/search tool spec i do wonder if it's better to have them in the core repo (we already have the ondemand loader tool in the core repo)